scrapy/signalmanager.py

```diff
@@ -23,7 +50,18 @@ def connect(self, receiver, signal, **kwargs):
         :type signal: object
         """
         kwargs.setdefault('sender', self.sender)
-        return dispatcher.connect(receiver, signal, **kwargs)
+        signal = self._ensure_signal(signal)
+        if not func_accepts_kwargs(receiver):
+            warnings.warn("The use of handlers that don't accept "
+                          "**kwargs has been deprecated, plese refer "
+                          "to the Signals API documentation.",
+                          ScrapyDeprecationWarning, stacklevel=2)
+            if receiver.__repr__() not in self._patched_receivers:
+                self._patched_receivers[receiver.__repr__()] = \
+                    lambda sender, **kw: _robust_apply(receiver, sender, **kw)
```

Comment:
Hmm, I think our solution here breaks the `weak=True` keyword argument functionality (since we're always keeping a strong reference in our `_patched_receivers` dictionary). Maybe it's enough to just mention this in the docs. (Since `weak=True` is a new feature and came together with requiring `**kwargs`.)